### PR TITLE
[Runtime] Add a function to check type creation

### PR DESCRIPTION
**Pull request by Azoy**
This adds a new SPI that takes a list of generic arguments and checks if said generic arguments are suitable to use as the generic arguments for the given type. If they are, gather all the extra arguments needed and fix up all of the pack pointers to call the metadata access function with these fixed up generic arguments.

### CODE REVIEW

**Comment by Azoy** [comment]
@swift-ci please test

**Comment by Azoy** [comment]
@swift-ci please test

File: include/swift/Runtime/Metadata.h [hunk 1]
```diff
@@ -1046,6 +1046,18 @@ void swift_initRawStructMetadata(StructMetadata *self,
                                  const TypeLayout *likeType,
                                  int32_t count);
 
+/// Check if the given generic arguments are valid inputs for the generic type
+/// context and if so call the access function to return
```
Comment (mikeash) [inline include/swift/Runtime/Metadata.h:1050]: The end of this comment is a bit unclear to me, suggest "and if so call the metadata access function and return the metadata."

**Review by mikeash — APPROVED** [review]

**Comment by Azoy** [comment]
@swift-ci please smoke test

**Pull request merged**